operator/v1/types_machineconfiguration.go

```diff
@@ -158,3 +179,170 @@ const (
 	// This feature only supports MAPI machinesets at this time.
 	MachineAPI MachineManagerMachineSetsAPIGroupType = "machine.openshift.io"
 )
+
+type NodeDisruptionPolicyStatus struct {
+	// clusterPolicies is a merge of cluster default and user provided node disruption policies.
+	// +optional
+	ClusterPolicies NodeDisruptionPolicyClusterStatus `json:"clusterPolicies"`
+}
+
+// NodeDisruptionPolicyConfig is the overall spec definition for files/units/sshkeys
+type NodeDisruptionPolicyConfig struct {
+	// files is a list of MachineConfig file definitions and actions to take to changes on those paths
+	// +optional
+	// +listType=map
+	// +listMapKey=path
+	// +kubebuilder:validation:MaxItems=50
+	Files []NodeDisruptionPolicyFile `json:"files"`
+	// units is a list MachineConfig unit definitions and actions to take on changes to those services
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=50
+	Units []NodeDisruptionPolicyUnit `json:"units"`
+	// sshkey maps to the ignition.sshkeys field in the MachineConfig object, definition an action for this
+	// will apply to all sshkey changes in the cluster
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey"`
```

Comment:
I think maybe you are thinking about the "Special" action type? We don't want the user to set that one. The only way out we could think of was having a different action type for Status and Spec, and making it such that Special is only settable on the Status side. Unfortunately, this would result in a lot of duplication, as this type is encapsulated through `NodeDisruptionPolicyAction` -> `NodeDisruptionPolicySSHKey`, `NodeDisruptionPolicyUnit`, `NodeDisruptionPolicyFile`  -> `NodeDisruptionPolicyClusterStatus` or `NodeDisruptionPolicyConfig`. I'm planning to do a follow-up commit for that, but wanted to run it by you in case there is another path?

Reply:
Generally we do advise having separate types for the spec and status to account for actions such as this, even though there would be duplication

The alternative is to use a CEL validation to prevent the special action, but that doesn't give you much freedom for future customisation nor does it make it easy to document the behaviour.

I'd recommend spec and status versions of the items